From 32d66aefc5aa1ff721f2dc5dc9a9b4d59870b89d Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Thu, 15 Jul 2021 19:37:22 +0300
Subject: [PATCH] [SW-0] ppboot: greco: remove sram scramble

In Greco, it is aligned with LKD team to do SRAM scrambling
when ARM Linux is booted. When Linux is booted, no FW components
are running from SRAM. This implementation is aligned with Gaudi1

Signed-off-by: Sailaja <sbandarupalli@habana.ai>
Change-Id: I2422490712e6d0bedcfeb8d809a59c59184ff8f5
---

diff --git a/arch/greco/arc_elf32/common.c b/arch/greco/arc_elf32/common.c
index 8a76d8b..72564d5 100644
--- a/arch/greco/arc_elf32/common.c
+++ b/arch/greco/arc_elf32/common.c
@@ -393,19 +393,6 @@
 	regmap_writel(val, mmPSOC_GLOBAL_CONF_AXI_DRAIN_CTRL);
 }
 
-/* initialize sram scrambling before preboot is downloaded */
-static void sram_scrambling_init(void)
-{
-	regmap_writel(1, mmDCON0_HBW_RTR_IF0_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON0_HBW_RTR_IF1_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON1_HBW_RTR_IF0_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON1_HBW_RTR_IF1_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON2_HBW_RTR_IF0_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON2_HBW_RTR_IF1_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON3_HBW_RTR_IF0_RTR_H3_SCRAM_SRAM_EN);
-	regmap_writel(1, mmDCON3_HBW_RTR_IF1_RTR_H3_SCRAM_SRAM_EN);
-}
-
 static void init_arc(void)
 {
 	halt_n_reset_arc(0);
@@ -436,8 +423,6 @@
 	init_arc();
 
 	axi_enable_drain();
-
-	sram_scrambling_init();
 
 	sw_all_rst_init();
 }
